### ISSUE TITLE
[BUG] gnome software

### ISSUE THREAD

**Issue by allaeddineomc** [fedora-silverblue/issue-tracker#191]
**Describe the bug**
gnome software have no chance to display any package on the first user account you get to create after the installation , it displays fine on the root account and all accounts created later , deleting all it's config and data files and folders in the first user home won't fix anything  

**To Reproduce**
Please describe the steps needed to reproduce the bug:
1. install and reboot
2. setup a user account
3. open gnome software
it won't display anything but updates work well

**Expected behavior**
shows apps to download

**OS version:**
```
Version: 34.20210817.0 (2021-08-17T00:48:48Z)
```

**Additional context**
BTW i have flathub enabled on my system

**Comment by TheOneandOnlyJakfrost** [comment]
When you first setup a system with the Silverblue installation media, you would need to point Gnome Software to the repositories like flathub, etc. as it defaults to having only the Fedora and Fedora updates repos enabled.

**Comment by allaeddineomc** [comment]
i did this and it didn't fixed the issue

**Comment by mcrha** [comment]
Do you know the gnome-software version in use, please? How did you enable the Flathub, in gnome-software or elsewhere? Some command line changes are not always recognized by the gnome-software, then it requires the gnome-software restart to notice them (`gnome-software --quit; gnome-software` , it's because the application runs in the background for the whole user session). When you search for a Flatpak application, will it be found? I would try for example "notes", or any other being provided by the Flathub.

I guess you do not see an error message like mentioned at https://github.com/fedora-silverblue/issue-tracker/issues/181 , do you?

**Comment by notklaatu** [comment]
I'm experiencing the same issue on a fresh Fedora 35 Silverblue install.

## Steps to reproduce:

1. Install Silverblue 35

2. Open GNOME Software. Go to the app menu > Software Repositories and confirm that Flathub repo is enabled.

![Screenshot from 2021-12-04 11-09-13](https://user-images.githubusercontent.com/2211671/144681193-152cc61e-6e5a-4d11-973c-2715fdfc200b.png)

3. Open Terminal to double-confirm:

```
$ flatpak remotes 
Name    Options
fedora  system,oci
flathub system,filtered
```

4. In GNOME Software, search for something on Flathub

![Screenshot from 2021-12-04 11-08-45](https://user-images.githubusercontent.com/2211671/144681449-a9950a32-eafc-4208-9f25-cd87d7217f54.png)
.
5. Search for something in a different flatpak repo, such as Emacs. Not found.

6. Open terminal:
```
$ flatpak search emacs
Name   Description                                                   Application ID    Version Branch Remotes
Devhe… A developer tool for browsing and searching API documentation org.gnome.Devhelp 41.2    stable fedora
$ flatpak search kdenlive
No matches found
```

## [Partial] solution:

1. Remove flathub repo

```
$ flatpak remote-delete flathub
```

2. Add flathub repo

```
$ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
```

3. Install something from flathub

```
$ flatpak install flathub org.kde.kdenlive
Looking for matches…
Required runtime for org.kde.kdenlive/x86_64/stable (runtime/org.kde.Platform/x86_64/5.15) found in remote flathub
Do you want to install it? [Y/n]: Y
[...]
```

4. Search for kdenlive in GNOME Software. Still fails.
5. Restart GNOME Software, search for kdenlive again. Still fails.
6. Search for anything on flathub. Fails.
7. Search for Emacs. Fails

## Expected results:

* GNOME Software should recognize the repositories listed in Software Repositories
* Flatpak should recognize repositories listed in `flatpak remotes

**Comment by allaeddineomc** [comment]
My workaround was to setup the first account after the first boot than create a new admin account than remove the first account

**Comment by mcrha** [comment]
> flathub system,filtered

The 'filtered' here means "not the whole Flathub is available, only a limited subset of the applications are offered".

> $ flatpak search emacs
> Name   Description                                                   Application ID    Version Branch Remotes
> Devhelp....

That a search for `emacs` returns DevHelp is kinda surprising for me. It can be the `flatpak` command line searches different set of properties than the Software, I guess.

> 2\. Add flathub repo
> ```
> $ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
> ```

As long as `flatpak remotes` doesn't say `filtered` for the `flathub`, you should be able to get the full Flathub content.

I tried this with gnome-software 41.1 and it cannot find DevHelp here, it works only after I restart it (`gnome-software --quit && gnome-software`). The change from https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1486 might help, but it is part of the 41.1; I guess it can be related to other code path.

**Comment by notklaatu** [comment]
I feel like gnome-software ought to alert the user that the repo is filtered, and how to unfilter it (is there an easier way other than removing it and then re-adding it?). From the user's perspective, this appears broken.

**Comment by mcrha** [comment]
> I feel like gnome-software ought to alert the user that the repo is filtered

I'm not aware of any such thing, but I do not know that code that much.

> (is there an easier way other than removing it and then re-adding it?)

You can just add it,flatpak itself handles it. You can also `flatpak remote-modify --system flathub --no-filter` (untested by me, see `flatpak remote-modify --help`)

**Comment by allaeddineomc** [comment]
As far as I'm concerned , the bug is gone on fedora 35 iso
